.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -6,12 +6,17 @@ on:
   workflow_dispatch:
     inputs:
       slug:
-        type: string
-        description: 'The slug of the plugin to deploy'
-      version:
-        type: string
-        description: 'The version of the plugin to deploy'
-
+        type: choice
+        default: 'webp-uploads'
+        options:
+        - audit-autoloaded-options
+        - audit-enqueued-assets
+        - dominant-color-images
+        - fetchpriority
+        - sqlite
+        - webp-support
+        - webp-uploads
+        description: 'Select a slug of the plugin to deploy'
```

Comment:
I don't believe that `inputs.{name}.options` supports dynamic assignment from a file, and even then we'd need to go further to only get the `slug` values.

However, we could add another workflow that modifies this one.

1. Run on every `push` to `trunk` that touches `plugins.json`.
2. Remove existing options from this workflow file.
3. Fetch, format and add the updated slugs to the deployment workflow file.
4. Commit to `trunk`.
5. Push to remote.

Something like this (Tested locally via ACT except for Commit and Push to remote):
```yaml
on:
  push:
    branches: trunk
    paths:
      - 'plugins.json'

jobs:
  update_slugs:
    name: Update slugs for deployment options
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3

      - name: Install jq
        run: |
          apt update -y
          apt install jq -y

      - name: Remove existing options
        run: sed -in '/options/, /description/{//!d;};' ./.github/workflows/test.yml

      - name: Fetch, format and add the updated slugs to the deployment workflow file
        run: |
          SLUGS=$(jq '"          - " + .[].slug' ./plugins.json | sed 's/"//g' )
          awk -v OPTIONS="options:\n$SLUGS" '{ sub( "options:", OPTIONS);print;}' ./.github/workflows/test.yml > tmp
          mv tmp ./.github/workflows/test.yml
          cat ./.github/workflows/test.yml

      - name: Commit to trunk
        run: |
          git add .
          git commit -m "Update standalone plugin slugs for deployment options."

      - name: Push to remote
        run: |
          GIT_BRANCH=${GITHUB_BASE_REF:-${GITHUB_REF#refs/heads/}}
          git fetch origin $GIT_BRANCH
          git push origin $GIT_BRANCH
  
```

Before:
```yaml
name: Manual workflow

on:
  workflow_dispatch:
    inputs:
      slug:
        type: choice
        options: 
          - test-plugin1
          - test-plugin3
        description: 'Select a slug of the plugin to deploy'

jobs:
  test:
    name: Run Test
    runs-on: ubuntu-latest
    steps:
      - name: A Test
        id: a-test
        run: echo "Howdy, admin!"
```

After:
```yaml
name: Manual workflow

on:
   workflow_dispatch:
     inputs:
       slug:
         type: choice
         options:
           - dominant-color-images
           - fetchpriority
           - webp-uploads 
         description: 'Select a slug of the plugin to deploy'

jobs:
  test:
    name: Run Test
    runs-on: ubuntu-latest
    steps:
      - name: A Test
        id: a-test
        run: echo "Howdy, admin!"
```

Reply:
Thanks both for the feedback. I don't think updating the workflow file is good idea 🤔 so i go with input filed option as @felixarntz suggested because GH did no allow dynamic value for inputs.